### ISSUE TITLE
row_log_interval / checkpoint_callback and early_stop_callback

### ISSUE THREAD

**Issue by david-waterworth** [PyTorchLightning/pytorch-lightning#3666]
I raised an issue in the past regarding the train score not being logged, which was resolved by setting `row_log_interval` less than the number of batches per epoch.

This also appears to interfere with callbacks. My model stopped writing checkpoints (other than the first) and early stopping also wasn't invoked. I think this occurred when I increased `batch_size` and didn't reduce `row_log_interval` accordingly. Worse was a single `epoch=0.ckpt` file was actually created but no subsequent one - if I understand correctly trainer.test() then restores this checkpoint rather than testing the trained model?

Is this the case or is something else going on here? If it is, it seems like there needs to be a mechanism to default `row_log_interval` to be at least once per epoch without the user having to manually calculate this? In particular `batch_size` is a hyperparameter and `row_log_interval` isn't (or shouldn't be) but it does (seem to) indirectly impact the model score if it prevents callbacks from being invoked?

**Comment by awaelchli** [comment]
True, I agree. row_log_interval should apply to global step and log every N steps regardless of epoch. #3466 

However, row_log_interval has no influence on callbacks, I am sure. It must be something else you are experiencing. For the callback, did you set the monitor and correctly? There were some bugfixes recently around checkpoint callback, check also the latest version.

**Comment by david-waterworth** [comment]
You're right - I tried again and I couldn't reproduce the issue.

What I did notice was I was getting the following (once I set verbose=True)

> RuntimeWarning: Early stopping conditioned on metric `val_loss` which is not available. Either add `val_loss` to the return of  validation_epoch end or modify your EarlyStopping callback to use any of the following: `val_early_stop_on`, `val_checkpoint_on`

This was a consequence of not refactoring correctly when I moved to the new pl.EvalResult(), my validation_step is now

        loss = self.loss_function(out, tgt_out)
        result = pl.EvalResult(checkpoint_on=loss, early_stop_on=loss) # added early_stop_on
        result.log('val_loss', loss)

Although now I am getting duplicate messages in my log

> Epoch 00013: val_checkpoint_on reached 0.50057 (best 0.50057), saving model to ./mlruns/1/6438eeab92ba4133a42086b0c9ab57bc/checkpoints/epoch=13.ckpt as top 1
> INFO:lightning:
> Epoch 00013: val_checkpoint_on reached 0.50057 (best 0.50057), saving model to ./mlruns/1/6438eeab92ba4133a42086b0c9ab57bc/checkpoints/epoch=13.ckpt as top 1

I think I'm setting the callbacks up correctly?

    # create trainer
    mlflow = pl.loggers.MLFlowLogger("Transformer")
    checkpoint_callback = pl.callbacks.ModelCheckpoint(verbose=True)
    early_stop_callback = pl.callbacks.EarlyStopping(verbose=True)
    trainer = pl.Trainer.from_argparse_args(args, gpus=[0], checkpoint_callback=checkpoint_callback, early_stop_callback=early_stop_callback, logger=mlflow, row_log_interval=1)

**Comment by awaelchli** [comment]
duplicate logging: yes, I can confirm this is likely just a weird python logging that we haven't configured correctly. work on this isn in progrss here: #3514 

We're currently sorting out some issues around results and checkpoint/earlystopping monitoring values. You can expect this to be fixed in v1.0. Until then, make sure that

1. You use the latest 0.9.1rc4 release
2. Have monitor="val_loss" in your modelcheckpoint (if you're passing a custom one into the Trainer)

The issue with early stopping monitor you mention should already be fixed. It is not using val_checkpoint_on key anymore.
